File: week_07/day_1/Try_Catch_Exceptions/code/CheckPoint_1_custom_exceptions/app/src/main/java/example/codeclan/com/checkpoint_1_custom_exceptions/ExampleTryCatch.java
package example.codeclan.com.checkpoint_1_custom_exceptions;

/**
 * Created by user on 30/08/2017.
 */

public class ExampleTryCatch{
    PetShop shop;

    public void run(){
        setup();
        Pet found = shop.findPetByName("Meowingtons");
        System.out.println("Found pet: " + found.getName());
    }

    public void setup(){
        shop = new PetShop();
        shop.addPet(new Dog("RoveR"));
        shop.addPet(new Cat("meowingtons"));
        shop.addPet(new Fish("FINlay"));
    }
}

File: week_10/c_sharp_course/week_1/day_5/try_catch/code/end_code/ExampleTryCatch.cs
using System;

public class ExampleTryCatch
{
  PetShop shop;

  public void Run()
  {
    this.Setup();
    try 
    {
      Pet found = shop.FindPetByName(null);
      Console.WriteLine("Found pet: " + found.GetName());
    }
    catch (NullStringException ex)
    {
      Console.WriteLine(ex.ToString());
      Console.WriteLine("Exception Message:");
      Console.WriteLine(ex.Message);
    }
    finally
    {
      Console.WriteLine("and finally...");
      Console.WriteLine("I'm done.");
    }
  }

  public void Setup()
  {
    shop = new PetShop();
    shop.AddPet(new Dog("Rover"));
    shop.AddPet(new Cat("MEOWingtons"));
    shop.AddPet(new Fish("FINlay"));
  }
}
